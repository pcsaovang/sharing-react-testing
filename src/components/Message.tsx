import classNames from "classnames";
import React from "react";

type MessageType = "success" | "warning" | "error";

type Props = {
  content: string;
  type?: MessageType;
};

const iconMap: Record<MessageType, React.ReactElement> = {
  success: (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      className="w-6 h-6 "
      fill="none"
      viewBox="0 0 24 24"
      stroke="currentColor"
    >
      <path
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        d="M5 13l4 4L19 7"
      />
    </svg>
  ),
  warning: (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      className="w-6 h-6"
      fill="none"
      viewBox="0 0 24 24"
      stroke="currentColor"
    >
      <path
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        d="M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-3L13.732 4c-.77-1.333-2.694-1.333-3.464 0L3.34 16c-.77 1.333.192 3 1.732 3z"
      />
    </svg>
  ),
  error: (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      className="w-6 h-6"
      fill="none"
      viewBox="0 0 24 24"
      stroke="currentColor"
    >
      <path
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
      />
    </svg>
  ),
};

function Message({ content, type = "success" }: Props): React.ReactElement {
  const containerClass = classNames(
    "flex",
    "items-center",
    "w-full",
    "px-2",
    "py-2",
    "rounded",
    {
      "text-green-700": type === "success",
      "bg-green-100": type === "success",
    },
    {
      "text-yellow-700": type === "warning",
      "bg-yellow-100": type === "warning",
    },
    {
      "text-red-700": type === "error",
      "bg-red-100": type === "error",
    }
  );

  return (
    <div className="flex justify-center mt-2">
      <div className={containerClass}>
        {iconMap[type]}
        <p className="ml-2 text-medium">{content}</p>
      </div>
    </div>
  );
}

export default Message;
