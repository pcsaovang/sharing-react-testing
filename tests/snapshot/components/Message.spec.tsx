import { render } from "@testing-library/react";
import Message from "../../../src/components/Message";

describe("Message component", () => {
  it("Render Message component with default props", () => {
    const { container } = render(<Message content="This is a message" />);
    expect(container).toMatchSnapshot();
  });

  it("Render Message component with type is warning", () => {
    const { container } = render(
      <Message content="This is a message" type="warning" />
    );
    expect(container).toMatchSnapshot();
  });

  it("Render Message component with type is error", () => {
    const { container } = render(
      <Message content="This is a message" type="error" />
    );
    expect(container).toMatchSnapshot();
  });
});
