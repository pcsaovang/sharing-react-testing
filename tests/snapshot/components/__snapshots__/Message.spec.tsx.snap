// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component Render Message component with default props 1`] = `
<div>
  <div
    class="flex justify-center mt-2"
  >
    <div
      class="flex items-center w-full px-2 py-2 rounded text-green-700 bg-green-100"
    >
      <svg
        class="w-6 h-6 "
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 13l4 4L19 7"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
      <p
        class="ml-2 text-medium"
      >
        This is a message
      </p>
    </div>
  </div>
</div>
`;

exports[`Message component Render Message component with type is error 1`] = `
<div>
  <div
    class="flex justify-center mt-2"
  >
    <div
      class="flex items-center w-full px-2 py-2 rounded text-red-700 bg-red-100"
    >
      <svg
        class="w-6 h-6"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
      <p
        class="ml-2 text-medium"
      >
        This is a message
      </p>
    </div>
  </div>
</div>
`;

exports[`Message component Render Message component with type is warning 1`] = `
<div>
  <div
    class="flex justify-center mt-2"
  >
    <div
      class="flex items-center w-full px-2 py-2 rounded text-yellow-700 bg-yellow-100"
    >
      <svg
        class="w-6 h-6"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-3L13.732 4c-.77-1.333-2.694-1.333-3.464 0L3.34 16c-.77 1.333.192 3 1.732 3z"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
      <p
        class="ml-2 text-medium"
      >
        This is a message
      </p>
    </div>
  </div>
</div>
`;
